src/components/views/rooms/MemberInfo.js

```diff
@@ -332,13 +332,42 @@ module.exports = withMatrixClient(React.createClass({
         });
     },
 
-    onMuteToggle: function() {
+    _warnSelfDemote: function() {
+        const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+        return new Promise((resolve) => {
+            Modal.createTrackedDialog('Demoting Self', '', QuestionDialog, {
+                title: _t("Warning!"),
+                description:
+                    <div>
+                        { _t("You will not be able to undo this change as you are demoting yourself, " +
+                            "if you are the last privileged user in the room it will be impossible " +
+                            "to regain privileges.") }
+                        <br />
+                        { _t("Are you sure?") }
+                    </div>,
+                button: _t("Continue"),
```

Comment:
I would generally label all the confirmation buttons with what the action is, ie. in this case, "Demote" rather than 'Continue'. It's a trivial way to add helpful information. See item 2 on https://uxplanet.org/5-essential-ux-rules-for-dialog-design-4de258c22116 (Likewise, the explanation here is good but the title and 'Are you sure?' could be better).